### PR TITLE
Port `amethyst_controls` to legion.

### PR DESCRIPTION
**Pull request by dkushner**
## Description

This PR continues work to integrate Legion ECS into Amethyst by porting the `amethyst_controls` module.

## Modifications

- Moved all existing systems definitions in `amethyst_controls` from struct-based to function-based using the Legion system builder. 

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Updated the content of the book if this PR would make the book outdated.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [ ] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [ ] Ran `cargo clippy --all --features "empty"`
- [ ] Ran `cargo test --all --features "empty"`

### CODE REVIEW

**Comment by jojolepro** [comment]
how is it going? do you need help on something? :3

**Comment by dkushner** [comment]
I'm opening this up for review to get some help/feedback on the implementation so far. I waffled between wanting to change up the entire system structure and sticking as close to the original implementation as possible. In the end, I tried to stick as close to the original implementation so that I could iteratively improve functionality. Things I need to fix:

[ ] Free movement system should apply the entity's rotation before translating for proper camera-relative movement.
[ ] Remove accumulated roll from free rotation system. 
[ ] Figure out why mouse look input and movement input are not executed simultaneously. 

There seems to be a number of places where the current implementation diverges with the "Legion way" of doing things, and I'm hoping that somebody can advise me on a more idiomatic way of accomplishing this.

**Comment by khionu** [comment]
Please note that a handful of minor bugs shouldn't be hard blockers. We would be happy to review lingering issues to see if we'd be happy to merge as is. While we would like to avoid regressions, this is part of a major change, and we can always try to fix them between merge and release.

File: amethyst_network/src/simulation/transport/udp.rs [hunk 1]
```diff
@@ -66,6 +66,10 @@ pub fn build_udp_network_send_system(
                                 delivery
                             ),
                         }
```
Comment (jojolepro) [inline amethyst_network/src/simulation/transport/udp.rs:68]: you'll need to pull the most recent version of legion_v2 and do a merge

**Comment by jojolepro** [comment]
superseded by https://github.com/amethyst/amethyst/pull/2378

**Pull request closed without merging**